src/frontend/src/optimizer/mod.rs

```diff
@@ -396,7 +396,10 @@ impl PlanRoot {
         plan = self.optimize_by_rules(
             plan,
             "Convert Distinct Aggregation".to_string(),
-            vec![UnionToDistinctRule::create(), DistinctAggRule::create()],
+            vec![
+                UnionToDistinctRule::create(),
+                DistinctAggRule::create(for_stream),
```

Comment:
https://github.com/risingwavelabs/risingwave/pull/8100#discussion_r1115214923

I'm going to do some evaluation on the cases where only one column is been distinct on. If in that case performance can also be improved by enable the backend distinct agg impl, then yes we can make this rule a pure batch rule.

Reply:
https://github.com/risingwavelabs/risingwave/pull/8100#discussion_r1115362057

Since we finally decide to maintain old behavior for queries w/o group by or w/ single distinct column, the rule won't be a no-op for streaming query then.